sa/sa.go

```diff
@@ -522,7 +548,18 @@ func (ssa *SQLStorageAuthority) UpdateRegistration(ctx context.Context, reg core
 	if err != nil {
 		return err
 	}
-	updatedRegModel.LockCol = rm.LockCol
+
+	if features.Enabled(features.AllowAccountDeactivation) {
+		erm := regType.(*regModelv2)
+		urm := updatedRegModel.(*regModelv2)
```

Comment:
`registrationToModel` has to return a `interface` so that it can handle both model types, because of this if we want to copy the `LockCol` over we have to cast the returned `interface` to the correct type and then reassign it to the `interface`.